File: linguagem_java/ex05.java
import java.util.Scanner;

public class ex05 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String mesAno;
        float ipca, menorIpca = Float.MAX_VALUE, maiorIpca = Float.MIN_VALUE, somaIpca = 0;
        String menorMesAno = "", maiorMesAno = "";
        int quantMeses = 0;

        System.out.println("Digite ano/mês e o IPCA (ou * para sair):");
        while (true) {
            mesAno = scanner.next();
            if (mesAno.equals("*"))
                break;

            ipca = scanner.nextFloat();

            if (ipca < menorIpca) {
                menorIpca = ipca;
                menorMesAno = mesAno;
            }
            if (ipca > maiorIpca) {
                maiorIpca = ipca;
                maiorMesAno = mesAno;
            }

            somaIpca += ipca;
            quantMeses++;
        }

        float mediaIpca = somaIpca / quantMeses;

        System.out.printf("%.2f (%s)\n", menorIpca, menorMesAno);
        System.out.printf("%.2f (%s)\n", maiorIpca, maiorMesAno);
        System.out.printf("%.2f\n", mediaIpca);

        scanner.close();
    }
}

File: linguagem_c/ex05.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_MESES 100

int main() {
    char mes_ano[8];
    float ipca, menor_ipca = 99999, maior_ipca = -99999, soma_ipca = 0;
    char menor_mes_ano[8], maior_mes_ano[8];
    int quant_meses = 0;

    printf("Digite ano/mês e o IPCA (ou * para sair):\n");
    while (1) {
        scanf("%s", mes_ano);
        if (mes_ano[0] == '*')
            break;

        scanf("%f", &ipca);

        if (ipca < menor_ipca) {
            menor_ipca = ipca;
            strcpy(menor_mes_ano, mes_ano);
        }
        if (ipca > maior_ipca) {
            maior_ipca = ipca;
            strcpy(maior_mes_ano, mes_ano);
        }

        soma_ipca += ipca;
        quant_meses++;
    }

    float media_ipca = soma_ipca / quant_meses;

    printf("%.2f (%s)\n", menor_ipca, menor_mes_ano);
    printf("%.2f (%s)\n", maior_ipca, maior_mes_ano);
    printf("%.2f\n", media_ipca);

    return 0;
}
